### PR TITLE
Hapistrano Deployment Integration

### PR DESCRIPTION
**Pull request by nickovivar**
# Description
A brief description of the changes.

# Reasons
A link to a GitHub and/or JIRA issue (if applicable).
Otherwise, a brief sentence about why you made these changes.

### CODE REVIEW

File: jekyll/_cci2/deployment-integrations.md [hunk 1]
```diff
@@ -507,6 +507,46 @@ If you would like to simplify your configuration workflows using a CircleCI orb
 
 For more detailed information about this orb, refer to the [CircleCI Google Cloud Orbs](https://circleci.com/orbs/registry/orb/circleci/gcp-cli) page in the [CircleCI Orbs Registry](https://circleci.com/orbs/registry/).
 
+## Hapistrano
+
+[Hapistrano](https://github.com/stackbuilders/hapistrano) is a deployment library based on Haskell similar to Ruby's Capistrano. Hapistrano (like Capistrano for Ruby) deploys applications to a new directory marked with a timestamp on the remote host. It creates this new directory quickly by placing a git repository for caching purposes on the remote server. Hapistrano uses a yml deploy file to perform tasks on the remote servers.
```
Comment (sestrella) [inline jekyll/_cci2/deployment-integrations.md]: @nickovivar this paragraph looks good, there just some minor suggestions:

```diff
- library based on Haskell
+ library written in Haskell
```

```diff
- similar to Ruby's Capistrano
+ similar to [Capistrano](https://capistranorb.com)
```

```diff
- Hapistrano (like Capistrano for Ruby) deploys applications to a new directory
+ Hapistrano deploys applications to a new directory 
```
```diff
- Hapistrano uses a yml deploy file to perform tasks on the remote servers.
+ Hapistrano image is available at [Docker Hub](https://hub.docker.com/r/stackbuilders/hapistrano)
```

File: jekyll/_cci2/deployment-integrations.md [hunk 2]
```diff
@@ -507,6 +507,46 @@ If you would like to simplify your configuration workflows using a CircleCI orb
...
+  deploy:
+    docker:
+      - image: stackbuilders/hapistrano:v0.3.9.2
+
+    steps:
+      - checkout
+      - deploy:
+          name: Deploy
+          command:
+            hap deploy -c deploy/deploy-file.yaml
```
Comment (sestrella) [inline jekyll/_cci2/deployment-integrations.md]: @nickovivar according to Hapistrano README.md the configuration file is usually named `hap.yaml`:

```diff
- hap deploy -c deploy/deploy-file.yaml
+ hap deploy -c hap.yaml
```

File: jekyll/_cci2/deployment-integrations.md [hunk 3]
```diff
@@ -507,6 +507,46 @@ If you would like to simplify your configuration workflows using a CircleCI orb
...
+
+```
+Here's an example of a deploy file.
```
Comment (sestrella) [inline jekyll/_cci2/deployment-integrations.md]: @nickovivar what do you think about?

```diff
- Here's an example of a deploy file
+ Hapistrano looks for a configuration file called `hap.yaml` that typically looks like this:
```

File: jekyll/_cci2/deployment-integrations.md [hunk 4]
```diff
@@ -507,6 +507,46 @@ If you would like to simplify your configuration workflows using a CircleCI orb
...
+
+```yaml
+---
+deploy_path: /path/to/deploy
```
Comment (sestrella) [inline jekyll/_cci2/deployment-integrations.md]: @nickovivar I think this code snippet looks good, although, I think there is still some room to improve by getting rid of some unnecessary values, and providing some concrete build steps related to Haskell:

```yaml
deploy_path: /path/to/app
host: example.com
repo: https://github.com/stackbuilders/hapistrano.git
revision: origin/master
build_script:
  - stack setup
  - stack build
restart_command: systemd restart app.service
```

**Comment by nickovivar** [comment]
@sestrella I Already made the changes. I think now is more concrete and provide more useful information.
Thanks!

**Review by sestrella — APPROVED** [review]
@nickovivar nice job, LGTM

**Pull request merged**